src/gh/components/DF_vizualization_settings/code.py

```diff
@@ -0,0 +1,60 @@
+#! python3
+
+import System
+import typing
+
+import Rhino
+import Rhino.Geometry as rg
+from ghpythonlib.componentbase import executingcomponent as component
+
+import Grasshopper as gh
+from Grasshopper.Kernel import GH_RuntimeMessageLevel as RML
+
+from diffCheck import df_vizualization
+
+class VisualizationSettings(component):
+    def RunScript(self,
+        i_value_type,
+        i_upper_threshold,
+        i_lower_threshold,
+        i_palette,
+        i_legend_height,
+        i_legend_width,
+        i_legend_plane,
+        i_histogram_scale_factor):
+
+        """
+        Compiles all the vizualization settings to feed to the vizualization component
+        """
+
+        # set default values
+        if i_palette is None: i_palette = "Jet"
```

Comment:
maybe for the `i_palette` we could just test if the name does not exists, either here or in the class `__init__()`:

https://github.com/diffCheckOrg/diffCheck/blob/b7607dc14c91fe2561d81962b362bc9b3a9c7a6c/src/gh/diffCheck/diffCheck/df_vizualization.py#L37-L68

Reply:
Added here also for i_value_type:

```
if i_palette not in  ["Jet", "Rainbow", "RdPu", "Viridis"]:
            ghenv.Component.AddRuntimeMessage(RML.Warning, "Possible values for i_palette are: Jet, Rainbow, RdPu, Viridis")
            return None
if i_value_type not in  ["dist", "RMSE", "MAX", "MIN", "STD"]:
            ghenv.Component.AddRuntimeMessage(RML.Warning, "Possible values for i_value_type are: dist, RMSE, MAX, MIN, STD")
            return None
```